### DISCUSSION TITLE
Discussions for private repositories

### DISCUSSION THREAD

**Discussion by HonzaMikula** [community/community#2853] [Discussions] [answered]
I like the new discussions, but frankly, they would be useful for private repositories as well. Is there a plan to allow them even for private repositories?

**Comment by shntnu** [comment]
This would be fantastic! We use GitHub for lab notebooks, and use [issues to discuss and report results](https://github.com/broadinstitute/cell-health/issues?q=is%3Aissue). 

Moving [issues to Discussions](https://github.com/broadinstitute/cell-health/discussions) would be great because it will create a much-needed separation between issues and discussions (and "closing" a discussion is a bit unnatural but we are forced to do it to declutter).

**Reply by evi-liu** [reply]
Which private repository would you like to enable Discussions on? We have a private beta available and are hoping to flag in a couple of repositories for feedback.

**Reply by shntnu** [reply]
Great!
https://github.com/jump-cellpainting/develop-computational-pipeline
https://github.com/jump-cellpainting/jump-cellpainting

**Reply by evi-liu** [reply]
I've flagged them both in! You can now toggle on Discussions in the repository settings under `Manage features`. Please let me know if you have any feedback.

**Reply by shntnu** [reply]
Excellent! Can you enable "Convert to discussion" for Issues? There are a bunch of discussions in issues right now :D

**Reply by evi-liu** [reply]
✅ done!

**Reply by JonathanAsbury-ACS** [reply]
Can you enable discussions on https://github.com/3mcloud/Serenity/
If possible, please enable it for the whole organization https://github.com/3mcloud
@evi-liu, 
We have been using it in our public repos and find it very valuable.  It's intuitive and easy to customize.  Great work!

**Reply by EXELANCE-LLC** [reply]
**I'm examining. I got some details now. I will compile and present it soon.**

**Comment by becca** [comment]
There is no official ETA yet for making Discussions available for private repositories. As soon as we have any information we'll be sharing it in the [public roadmap](https://github.com/github/roadmap)!

**Reply by okproject** [reply]
Hi, roadmap link is broken. Here is the fixed one : [https://github.com/github/roadmap](https://github.com/github/roadmap)

**Reply by cadop** [reply]
@becca does the lack of ETA mean it will be at least 6 months (looking at the roadmap to Q2 2021 its not listed), or that its not even being considered (since it doesn't show in the Future section either)?  

This really would be a great feature for academics/research/teaching.  On the surface it seems to be trivial to enable it on private repos. Maybe not appropriate to ask, but is this simply a business decision with overlapping on the Teams discussion ability?

**Reply by evi-liu** [reply]
It's definitely a consideration! I'd love to get some feedback on my [comment below](https://github.com/github/feedback/discussions/2853#discussioncomment-263636) so we can get the ball rolling sooner rather than later for you all :).

**Comment by sjengle** [comment]
This feature for private repositories would be amazing for use in classrooms!

**Reply by kristofdombi** [reply]
This would also serve well for asynchronous, remote teams to discuss ideas/problems of a given codebase. 🙂 Personally, can't wait to enable this for our team! 🤓

**Reply by evi-liu** [reply]
Hi @sjengle and @kristof0425! Which private repositories would you like to enable Discussions on? We have a private beta available and are hoping to flag in a couple of repositories for feedback.

**Reply by kristofdombi** [reply]
Sorry, @evi-liu, I was on a holiday this week, didn't notice your message! We have [this](https://github.com/kinsta/mykinsta16) private repository, where it'd be really beneficial to have the Discussions feature enabled.

**Comment by evi-liu** [comment]
Hi all, PM on Discussions here. Thanks for all the feedback. I'd love to start the conversation for enabling Discussions for private repositories. To start, are there any features you'd need on Discussions with private repositories that currently aren't available on Discussions for public repositories?

**Reply by cadop** [reply]
Just the same thing would be a nice start. 

I think there are other things that would be helpful globally to the interface though. For example, easier ways to see/tag/deal with Milestones and Projects. If discussing in a thread about some milestone, or attaching some comments to one, etc. without having to make a new issue for a milestone, it would be a nicer way to integrate instead of dealing with slack for project planning+github for just entering the project plan.

**Reply by HonzaMikula** [reply]
Probably nothing huge. Frankly, I would like to use Discussions in GH for capturing mainly non-technical documentation since GH wiki is far from great (technical docs we have directly in repo). So, probably what would help me to have some ways how to organize those "answered" items since naturally the voting and marking discussion items as answered are nice for public repos with many people in the community, but in the private repo is not that needed. It's more for the team to have ways to organize it as they need.

**Reply by evi-liu** [reply]
>  For example, easier ways to see/tag/deal with Milestones and Projects. 

Would this only be valuable on your private repositories? Or on your public repositories, too?

**Reply by cadop** [reply]
That would be useful for both. 

I guess a private-specific feature that would be nice is a chat feature (although this could also be useful for public). Similar to how long comments on SE get moved to chat.  The Discussions in the way it's formatted seems like a clone of stackoverflow, which works well for QA type discussions, but takes a lot of space and is hard to search through for longer conversations.  

Something like project planning and group updates would benefit from a modified format that doesn't have as much spacing between each response and uses more horizontal real-estate.  It would also be nice to have a threaded reply that is easily collapsed, like slack or discord. I think the end-goal here would be to get rid of slack so we could do project _planning_ in github, not just document progress on the project. 

I think this type of discussion ability is helpful and makes sense for private repos since it is controlled and you don't get tons of spam. 

For public it might be a bit more chaotic. On the other hand, it may be nice for some public repos, e.g., it's annoying to look through 1k issues (https://github.com/numba/numba) and then find a totally other website for chatting (https://gitter.im/numba/numba).

**Reply by vanbroup** [reply]
Private discussions could just provide the same functionality as public.
I would love the have discussions on private repositories rather sooner than later.

**Reply by evi-liu** [reply]
@vanbroup do you mind expanding on your use case for private discussions? what types of repositories are you using these on that can't be public?

**Reply by surjithctly** [reply]
@evi-liu  I think the only limitation would be whoever has write / read access to that repository should be able to access the Discussions. That's obvious though :)

Another option I would like to see is a minimal header option while we are Discussions tab? maybe Admins can set minimal header option (only with GitHub logo and some menus and a back button) instead of 180px height header with 30+ clicks & actions. 

Also in Public Repository, It would be great if there is an option for admins to set "Join the Discussions" banner before letting them post anything. So that we can see Total Members Count (in Sidebar) + Send them updates on Each Post (like subscribing)

**Reply by heath3n** [reply]
@evi-liu Something I definitely want to see is the ability to have public discussions for private repositories; the current solution suggested by Github of having an empty public repository dedicated to issues is annoying. Also, having the ability to make projects public would be great too. I dislike having to use an external Trello board linked to my private repo to provide visibility to non-contributing community members.

**Reply by vanbroup** [reply]
It would be useful if you could start a new discussion by email, for example if you want to forward an external announcement to be discussed. This is what currently happens a lot on our discussion lists.

Only members of the private repository should be allowed to create new discussions by email, they could use an email address like: `{category}.{project}.{organization}@github.com` or `{project}.{organization}@github.com` for the general category.

**Comment by vanbroup** [comment]
In general I would state that many discussions happen in private before
products, solutions or dicisions are released to the public.

We have closed user groups to discuss product issues, ideas, industry
developments, policies, practices, etc.

These groups can consist of employees, customers and other relevant
stakeholders. For some projects they have 'code' or documentation access
for others this would just to gather feedback.

**Reply by vanbroup** [reply]
@evi-liu looks like email replies are processed as a new answer instead of a reply to the actual comment.

**Comment by s-mark-holmes-ii** [comment]
@evi-liu Could you enable discussions on this private repo? https://github.com/iheartradio/web

**Reply by evi-liu** [reply]
enabled! you can now toggle on Discussions from the repository settings under `Manage features`

**Reply by s-mark-holmes-ii** [reply]
@evi-liu You are the MVP! Thanks!

**Comment by evi-liu** [comment]
### Private beta available!

We have a private beta available and are hoping to flag in a couple more repositories for feedback! If you're interested, please reply to this thread with the repository. Limiting to the first 10 commenters for now.

**Reply by surjithctly** [reply]
Please enable for this repo: https://github.com/web3templates/community/

**Reply by kristofdombi** [reply]
Hello 👋 
Please enable it for us!
https://github.com/kinsta/mykinsta16

Thank you!

**Reply by HonzaMikula** [reply]
Please this one: https://github.com/salsita/adco. Thank you.

**Reply by ImSkully** [reply]
Hey there! https://github.com/ImSkully/emeraldgaming - Please and thank you! 👍

**Reply by cadop** [reply]
> ### Private beta available!
> We have a private beta available and are hoping to flag in a couple more repositories for feedback! If you're interested, please reply to this thread with the repository. Limiting to the first 10 commenters for now.

Could I get it enabled on this? https://github.com/cadop/design_perception  thanks!

**Reply by pitmonticone** [reply]
Please enable for these repositories 

* https://github.com/InPhyT/ComputationalEpidemiologyProject 
* https://github.com/InPhyT/Econophysics

Thanks.

**Reply by evi-liu** [reply]
✅ All repositories in this thread have been flagged into the private beta, so an admin/maintainer may now toggle on Discussions in the repository settings under `Manage features`.

Please respond to this thread with any feedback, particularly revolving Discussions capabilities needed in private repos that are not currently available for public repositories. Thanks!

**Reply by maplesteve** [reply]
@evi-liu If there's a spot left for another private repo, it would be great if you could add us to the list. Currently we're using the (issue based) "discussion" on the team level, but the new product would suit our needs much more and we would be happy to move to it.
- https://github.com/frnde/funk-meta

Thanks!

**Reply by robertwbradford** [reply]
Hello, if there is still an opening can I request the beta for the following repos?
- https://github.com/ICSEng/alm-connect
- https://github.com/ICSEng/eden
- https://github.com/ICSEng/jsstack-starter

Thank you!

**Reply by maplesteve** [reply]
Hi @evi-liu , would it be possible to join the beta for private Repos? Maybe my [previous](https://github.com/github/feedback/discussions/2853#discussioncomment-338536) "request" got buried in your list. The repo would be https://github.com/frnde/funk-meta which would then become the collaboration hub for our team.
Thanks!!

**Reply by esatterwhite** [reply]
@evi-liu We would love to try discussions internally, can you enable this one
https://github.com/answerbook/buzzsaw

**Reply by evi-liu** [reply]
✅ All repositories in this thread have been flagged into the private beta, so an admin/maintainer may now toggle on Discussions in the repository settings under Manage features.

**Reply by dhiguero** [reply]
@evi-liu If you have more invites, we are eager to test this internally on https://github.com/napptive/product

**Reply by munderseth** [reply]
Hi @evi-liu we would also appreciate the chance to use it would with one of our private repos - https://github.com/s2technologies/testspace.test.man. 
 - We are also interested in future API support
 - And we have some ideas about the possible integration of *Discussions* with our GitHub Testspace app concerning Exploratory Testing (new feature coming).

**Reply by leandrogrillo** [reply]
Hi @evi-liu could you enable for these repos:

- https://github.com/leroy-merlin-br/front-end-place/
- https://github.com/leroy-merlin-br/backyard/

**Reply by mwz** [reply]
Hi @evi-liu, if you have any more invites available I'd like to try out private discussions in [ovotech/discussions](https://github.com/ovotech/discussions). We have over 600 people in ovotech, so hopefully we should be able to get back to you with some feedback 😄

**Reply by TimothyIp** [reply]
Hi @evi-liu , the timing couldn't be more perfect! We are currently shopping around for ways to have a place for proper discussions for our eng department. If there are more invites *fingers crossed*, we'd like to give it a try at https://github.com/RoseRocket/roserocket

We will be happy to provide feedback :)

**Reply by bartr** [reply]
Could you enable for: https://github.com/retaildevcrews/links please?

tia!
b

**Comment by JonathanAsbury-ACS** [comment]
Please enable discussions on https://github.com/3mcloud/Serenity/
If possible, please enable it for the whole organization https://github.com/3mcloud
I read that the ability to apply it to the Org is currently not possible and added a request to include it.  
Do you offer visibility into what functionality is scoped and prioritized given the Road Map is very high-level?  

We have been using it in our public repos and find it very valuable. It's intuitive and easy to customize. Great work!

**Comment by beperev** [comment]
Hello,

That is nice that discussions are available for private repositories. 

I used to provide my students assignments by using template repositories on GitHub as starter code for an assignment on GitHub Classroom. In this way, my students can clone it and obtain their own private repository for the assignment. 

I would like to have a discussion for each assignment so that all students can have space to ask their questions regarding the assignment at hand and receive answers from classmates and educators. That is, my idea is that all my students have access to the same discussion feature to discuss any problem that arose when developing their programs regarding the same assignment (that is, one discussion per assignment and not per each student's private repository created for the assignment). I have tried to add the Discussion feature to a Github template, thinking that perhaps the cloned repositories could see the discussions created but I am not able to create a new discussion in a template repository (I do not see any option).

I have tried to do this with issues but they are linked to concrete repositories.

Any idea? perhaps there is another GitHub utility that allows me to do that?

Thank you very much!

**Reply by cadop** [reply]
There is a canvas integration with github, and also github education that may help https://education.github.com/

**Reply by beperev** [reply]
I do not see any option.

Perhaps a solution would be to be able to create new discussions associated to the template repository?

**Comment by evi-liu** [comment, marked as answer]
Hi all, as many of you have probably seen by now, we released [Discussions for private repositories ](https://github.blog/2021-03-09-github-discussions-now-available-for-private-repositories/) yesterday 🎉 ! Go ahead and toggle it on for your repositories now.

**Reply by cadop** [reply]
Thank @evi-liu , glad this really made it into the product!

**Reply by jsunsawyer** [reply]
@evi-liu I just got a notification about this in one of my private repos, but when I enable it, I get the error message `You can't perform that action at this time.` at the top of the page.

Also, I don't have a `New Discussion` button, so I can't really test things out, as I can't start a discussion.

Edit: To be clear, discussions appear to be enabled. I have the discussions dashboard in my repo. I just have an error message and the missing button.

**Reply by jrstanley** [reply]
@jsunsawyer I see the same. However, if I create an Issue then I am able to convert that into a Discussion.

**Comment by aleksandrzhiliaev** [comment]
Hey, thanks for the Discussions feature. Looks really promising. Is there any way to have Discussions API for a specific repository?
Thanks!
